### PR TITLE
[node] tidied up and de-duplicated tls related options

### PR DESCRIPTION
**Pull request by hoo29**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/dist/latest-v8.x/docs/api/tls.html

- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/node/v8/index.d.ts [hunk 1]
```diff
@@ -4816,15 +4805,15 @@ declare module "tls" {
              * An array of strings or a Buffer naming possible NPN protocols.
              * (Protocols should be ordered by their priority.)
              */
-            NPNProtocols?: string[] | Buffer,
+            NPNProtocols?: string[] | Buffer[] | Uint8Array[] | Buffer | Uint8Array;
```
Comment (Flarna) [inline types/node/v8/index.d.ts]: I think line should be ended with `,` here, not `;`

File: types/node/v8/index.d.ts [hunk 2]
```diff
@@ -4816,15 +4805,15 @@ declare module "tls" {
...
             /**
              * An array of strings or a Buffer naming possible ALPN protocols.
              * (Protocols should be ordered by their priority.) When the server
              * receives both NPN and ALPN extensions from the client, ALPN takes
              * precedence over NPN and the server does not send an NPN extension
              * to the client.
              */
-            ALPNProtocols?: string[] | Buffer,
+            ALPNProtocols?: string[] | Buffer[] | Uint8Array[] | Buffer | Uint8Array;
```
Comment (Flarna) [inline types/node/v8/index.d.ts]: I think line should be ended with `,` here, not `;`

File: types/node/v8/index.d.ts [hunk 3]
```diff
@@ -4834,7 +4823,7 @@ declare module "tls" {
              * SecureContext.) If SNICallback wasn't provided the default callback
              * with high-level API will be used (see below).
              */
-            SNICallback?: (servername: string, cb: (err: Error | null, ctx: SecureContext) => void) => void,
+            SNICallback?: (servername: string, cb: (err: Error | null, ctx: SecureContext) => void) => void;
```
Comment (Flarna) [inline types/node/v8/index.d.ts]: I think line should be ended with `,` here, not `;`

File: types/node/v8/index.d.ts [hunk 4]
```diff
@@ -4900,7 +4889,7 @@ declare module "tls" {
          * @param callback - callback(err) will be executed with null as err, once the renegotiation
          * is successfully completed.
          */
-        renegotiate(options: TlsOptions, callback: (err: Error | null) => void): any;
+        renegotiate(options: { rejectUnauthorized: boolean, requestCert: boolean }, callback: (err: Error | null) => void): any;
```
Comment (Flarna) [inline types/node/v8/index.d.ts]: `rejectUnauthorized` and `requestCert` are. optional

**Review by Flarna — CHANGES_REQUESTED** [review]
Would be nice to apply the changes also to NodeJs 9 typings. In 9.3.0 a optional `clientCertEngine: string` was added to `TlsServerOptions` (see https://nodejs.org/dist/latest/docs/api/tls.html#tls_tls_createserver_options_secureconnectionlistener).

File: types/node/v8/index.d.ts (hunk 4 above)
Comment (hoo29) [reply types/node/v8/index.d.ts]: Thanks, that could have been bad!

**Comment by hoo29** [comment]
I updated node v9 and didn't use Pick so I didn't have to put typescript 2.1 under pretty much every type (a  rabbit hole I started to go down). Can anyone explain why I am getting lots of messages to put '// TypeScript Version: 2.1' just under the header?

**Comment by Flarna** [comment]
I think the reason is that Lookup types need also 2.1 similar as `Pick` see https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-1.html

I fear the solution here is copy paste... I can remember a similar topic a while ago in url area.

In CI a check is done that dependent modules requires the same or a higher TS version. As usually all modules depend on latest Node.JS typings the issue popped up once you updated v9.
But I would keep v8 and v9 in sync here as people out their may lock Node to v8 as it is LTS.

**Comment by hoo29** [comment]
Thank you for the help! Copied and pasted it for now. Hopefully, the default TS version will be updated soon.

File: types/node/v8/index.d.ts [hunk 5]
```diff
@@ -17,7 +17,9 @@
 //                 Sebastian Silbermann <https://github.com/eps1lon>
 //                 Hannes Magnusson <https://github.com/Hannes-Magnusson-CK>
 //                 Alberto Schiabel <https://github.com/jkomyno>
+//                 Huw <https://github.com/hoo29>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
```
Comment (Flarna) [inline types/node/v8/index.d.ts]: I think we should keep NodeJs v8 on the same TS version then latest.

**Review by dex4er — APPROVED** [review]
smtp-server: ok

**Review by markisme — APPROVED** [review]

**Review by Flarna — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
Thank you!

**Pull request merged**